schema_based_on_diagram.sql

```diff
@@ -0,0 +1,44 @@
+CREATE TABLE patients (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    name VARCHAR(50) NOT NULL,
+    date_of_birth DATE NOT NULL
+);
+
+CREATE TABLE medical_histories (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    admitted_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL,
+    status VARCHAR(50) NOT NULL
+);
+
+ALTER TABLE medical_histories 
+ADD FOREIGN KEY (patient_id) REFERENCES patients (id);
+
+
+CREATE TABLE treatments (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    type VARCHAR(20) NOT NULL,
+    name VARCHAR(50) NOT NULL
+);
+
+CREATE TABLE invoices (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    total_amount DECIMAL(5,2) NOT NULL,
+    generated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL,
+    payed_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
+);
+
+ALTER TABLE invoices
+ADD FOREIGN KEY (medical_history__id) REFERENCES medical_histories (id);
+
+CREATE TABLE invoice_items (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    unit_price DECIMAL(2,2) NOT NULL,
+    quantity INT NOT NULL,
+    total_price DECIMAL(6,2) NOT NULL
+);
+
+ALTER TABLE invoice_items
+ADD FOREIGN KEY (invoice_id) REFERENCES invoices (id);
+
+ALTER TABLE invoice_items
```

Comment:
- [ ] Please make sure to create the "join table" for the many-to-many relationships as requested in the project requirements.

![image](https://user-images.githubusercontent.com/83803414/217532826-2fd1cfb5-ff76-49b6-9ce8-fcb07ea0482a.png)